core/trino-main/src/main/java/io/trino/execution/QueryIdGenerator.java

```diff
@@ -85,7 +87,7 @@ public synchronized QueryId createNextQueryId()
         if (counter > 99_999) {
             // wait for the second to rollover
             while (MILLISECONDS.toSeconds(nowInMillis()) == lastTimeInSeconds) {
-                Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
+                sleepUninterruptibly(1, SECONDS);
```

Comment:
can you also update the `// only generate 100,000 ids per day` comment? (assuming it should be "per second")

Reply:
The comment is correct. The counter only rolls at the day boundary, not every second. Otherwise, the counter would usually be `00000`. But when the limit is reached, we have to wait for the timestamp to change to reset the counter, to avoid duplicates. (in the very unlikely case we generated all 100,000 in the previous second)

Reply:
`Wait for the timestamp to change to reset the counter to avoid duplicates in the very unlikely case we generated all 100,000 in the previous second` sounds like the perfect code comment here (albeit a bit long).

I was confused too.

Reply:
Good idea. I added that and updated some other comments to be more clear